### PR TITLE
Remove some warnings

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
This'll clean up some of the warnings that CI is giving right now. There are still quite a few places that can be fixed up that aren't being reported on right now, but that can be done as part of a later clean-up.

### CODE REVIEW

File: src/pyjion/absint.h [hunk 1]
```diff
@@ -330,32 +330,32 @@ class AbstractInterpreter {
...
+    Label getOffsetLabel(size_t jumpTo);
+    void forIter(size_t loopIndex);
+    void forIter(size_t loopIndex, AbstractValueWithSources* iterator);
 
     // Checks to see if we have a null value as the last value on our stack
     // indicating an error, and if so, branches to our current error handler.
-    void errorCheck(const char* reason = nullptr, int curByte = -1);
-    void intErrorCheck(const char* reason = nullptr, int curByte = -1);
+    void errorCheck(const char* reason = nullptr, size_t curByte = ~0);
+    void intErrorCheck(const char* reason = nullptr, size_t curByte = ~0);
```
Comment (DanielRosenwasser) [inline src/pyjion/absint.h:344]: I wasn't sure what to do about these and `branchRaise` below. `~0` seemed to be the most semantically consistent.

File: src/pyjion/absint.cpp [hunk 2]
```diff
@@ -1302,7 +1302,7 @@ void AbstractInterpreter::branchRaise(const char *reason, size_t curByte) {
         m_comp->emit_trace_exception();
 
     // number of stack entries we need to clear...
-    int count = m_stack.size() - entryStack.size();
+    auto count = m_stack.size() - entryStack.size();
```
Comment (DanielRosenwasser) [inline src/pyjion/absint.cpp:1305]: OOPS

**Comment by DanielRosenwasser** [comment]
@tonybaloney I think I've cleared out most (if not all) of the CI warnings, and tests are passing locally. When you get the chance, I'd like to get some feedback to see if I'm way off course.

**Comment by tonybaloney** [comment]
This looks great for absint.cpp, I think next to look at `ilgen` and update the interfaces to standard 32-bit and 64-bit types. Especially the `ld_i()` and `ld_i4()` methods.

That should be done before tackling `pycomp.h/cpp/ipycomp.h`. 

Once the IL generation API is on standard types, the pycomp functions can be changed and that'd fix the remaining warnings.

Also I bet #118 and #204 are related to this

**Pull request merged**